### PR TITLE
Add method to post annotations for project

### PR DESCRIPTION
**Pull request by lewfish**
This PR adds a method to posts annotations for a project that are stored in a JSON file on S3. This will be called by the Lambda function in https://github.com/azavea/raster-foundry-deployment/pull/115. This also makes some other changes to stay in sync with the latest RV and removes some methods which are not useful and annoying to maintain.

To test making a prediction, run the following commands:
```
from rasterfoundry.api import API
token = 'XXX'
api = API(refresh_token=token)

from rasterfoundry.utils import RasterVisionBatchClient

rv_batch_client = RasterVisionBatchClient(
    branch_name='lf/rfint', job_queue='raster-vision-cpu',
    job_definition='raster-vision-cpu')

from rasterfoundry.models.project import Project

uuid = 'de3780a1-9f3e-4fa3-a811-ad3060b64866'
proj = api.client.Imagery.get_projects_uuid(uuid=uuid).result()
proj = Project(proj, api)

inference_graph_uri = 's3://raster-vision-lf-dev/detection/trained-models/cowc-potsdam/5cm/inference-graph.pb'
label_map_uri = 's3://raster-vision-lf-dev/detection/configs/label-maps/cowc.pbtxt'
predictions_uri = 's3://raster-vision-lf-dev/detection/predictions/cowc-potsdam/5cm/de3780a1-9f3e-4fa3-a811-ad3060b64866.json'

proj.start_predict_job(rv_batch_client, inference_graph_uri, label_map_uri,
                       predictions_uri, channel_order=[0, 1, 2])
```

To test posting predictions, run the following after clearing the annotations in the UI at https://app.rasterfoundry.com/projects/edit/de3780a1-9f3e-4fa3-a811-ad3060b64866/annotate
```
proj.post_annotations(predictions_uri)
```

Connect #41

### CODE REVIEW

**Comment by lewfish** [comment]
Note: I need to make a release containing this PR.

File: rasterfoundry/spec.yml [hunk 1]
```diff
@@ -1,100 +1,149 @@
...
   - application/json
 consumes:
   - application/json
-
 x-top-matter:
   - title: Introduction
     level: 1
-    content: |
-      The Raster Foundry API allows developers to find, view, and analyze geospatial data in Raster Foundry. As the name implies, it's especially useful for working with large raster datasets like satellite imagery.
+    content: >
```
Comment (Lknechtli) [inline rasterfoundry/spec.yml]: pretty sure this will break things in the spec - the swagger.io editor defaults to swagger 3.0, and we're using 2.0.

Comment (lewfish) [reply rasterfoundry/spec.yml]: Fixed it. I'll use the 2.0 Editor for now on.

**Comment by lewfish** [comment]
Can we try to get this reviewed soon? It's blocking me on https://github.com/azavea/raster-foundry-deployment/pull/115

File: rasterfoundry/settings.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
 RV_CPU_QUEUE = 'raster-vision-cpu'
 RV_CPU_JOB_DEF = 'raster-vision-cpu'
-RV_PROJ_CONFIG_DIR_URI = 's3://raster-vision-od/configs/projects'
+RV_PROJ_CONFIG_DIR_URI = 's3://raster-vision-lf-dev/detection/configs/projects/rf-generated'
```
Comment (Lknechtli) [inline rasterfoundry/settings.py]: shouldn't these kinds of settings be stored in a gitignored file with an example placeholder file committed? We do something similar in RF

Comment (lewfish) [reply rasterfoundry/settings.py]: Good point. I have too much other stuff do right now, so I made an issue to do that: https://github.com/raster-foundry/raster-foundry-python-client/issues/43

**Pull request merged**